### PR TITLE
Make `SkipReason` properties settable

### PR DESCRIPTION
**Pull request by dougbu**
Also update `dnx451` dependencies to match current reality
- avoid warnings about incorrect `System.Runtime` versions

### CODE REVIEW

File: src/Microsoft.AspNet.Testing/xunit/FrameworkSkipConditionAttribute.cs [hunk 1]
```diff
@@ -23,15 +23,9 @@ public bool IsMet
...
         }
 
-        public string SkipReason
-        {
-            get
-            {
-                return "Test cannot run on this runtime framework.";
-            }
-        }
+        public string SkipReason { get; set; } = "Test cannot run on this runtime framework.";
```
Comment (rynowak) [inline src/Microsoft.AspNet.Testing/xunit/FrameworkSkipConditionAttribute.cs]: :clap:

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Testing/project.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "frameworks": {
         "dnx451": {
             "frameworkAssemblies": {
-                "System.Runtime": "",
-                "System.Reflection": "",
-                "System.Threading.Tasks": ""
+                "System.Runtime": "4.0.10.0",
```
Comment (davidfowl) [inline src/Microsoft.AspNet.Testing/project.json]: Remove these

File: src/Microsoft.Dnx.TestHost/project.json [hunk 3]
```diff
@@ -31,7 +31,7 @@
     "frameworks": {
         "dnx451": {
             "frameworkAssemblies": {
-                "System.Runtime": "4.0.0.0",
```
Comment (davidfowl) [inline src/Microsoft.Dnx.TestHost/project.json]: remove these versions

File: src/Microsoft.Framework.Logging.Testing/project.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
         },
         "dnx451": {
             "frameworkAssemblies": {
-                "System.Runtime": "4.0.0.0"
```
Comment (davidfowl) [inline src/Microsoft.Framework.Logging.Testing/project.json]: Remove these verisons

File: src/Microsoft.Framework.WebEncoders.Testing/project.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
         },
         "dnx451": {
             "frameworkAssemblies": {
-                "System.Runtime": "4.0.0.0"
+                "System.Runtime": "4.0.10.0"
```
Comment (davidfowl) [inline src/Microsoft.Framework.WebEncoders.Testing/project.json]: Remove these versions

**Comment by pranavkm** [comment]
:shipit: when @davidfowl is happy.

File: src/Microsoft.Framework.WebEncoders.Testing/project.json (hunk 5 above)
Comment (dougbu) [inline src/Microsoft.Framework.WebEncoders.Testing/project.json]: So the empty string is correct for `dnx451` framework assemblies? Same for `net45` (e,g, just above)?

**Comment by dougbu** [comment]
@davidfowl updated

**Comment by davidfowl** [comment]
:shipit: When green

**Comment by dougbu** [comment]
d8a6507

**Pull request closed without merging**